api/api/services/sendgrid_mail.py

```diff
@@ -24,11 +24,13 @@ class SendgridClient:
     def __init__(self, recipient: str, sender: Optional[str] = SendgridParameters.DEFAULT_SENDER):
         self.logger = logging.getLogger(__name__)
         logging.basicConfig(level=logging.INFO)
-
         self.sendgrid_client = SendGridAPIClient(
             api_key=SendgridParameters.SENDGRID_API_KEY)
         self.mail = Mail(sender, To(recipient), subject=None)
-        self.mail.template_id = SendgridParameters.TEMPLATE_ID
+
+    def set_template_id(self, template_id: str):
+        """Set the template ID for the email"""
+        self.mail.template_id = template_id
```

Comment:
this function just sets something in the class, why not doing directly ? this function is not required